tests/contrib/codemeta/test_codemeta_custom_fields.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""CodeMeta custom fields tests."""
+
+import pytest
+from invenio_access.permissions import system_identity
+
+from invenio_rdm_records.cli import (
+    create_records_custom_field,
+    custom_field_exists_in_records,
+)
+from invenio_rdm_records.contrib.codemeta.custom_fields import (
+    CODE_CUSTOM_FIELDS,
+    CODE_NAMESPACE,
+)
+from invenio_rdm_records.proxies import (
+    current_rdm_records_service as rdm_records_service,
+)
+from invenio_rdm_records.records import RDMDraft, RDMRecord
+
+
+def create_record(record_id):
+    """Uses rdm records service to publish a record."""
+    # Publish a record
+    return rdm_records_service.publish(system_identity, record_id)
+
+
+def create_draft(draft_data):
+    """Uses rdm records service to create a draft."""
+    return rdm_records_service.create(system_identity, draft_data)
+
+
+@pytest.fixture(scope="function")
+def codemeta_fields_record(minimal_codemeta_record):
+    """Creates a record containing CodeMeta fields."""
+    draft = create_draft(minimal_codemeta_record)
+    record = create_record(draft.id)
+    return record
+
+
+def test_custom_fields_loading(running_app, cli_runner, codemeta_cf_name):
+    """Tests whether custom fields are correctly loaded on demand."""
+    # If custom fields are not initialised, "custom_field_exists_in_records" fails with KeyError['properties']
```

Comment:
Is this a bug? It breaks here https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/custom_fields/mappings.py#L49

The function `field_exists` assumes that `mapping['properties']` exists.

Reply:
It assumes `properties` exists because is how the lower `Mapping` class is implemented, see <https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/helpers/mapping.py#L106>

If `properties` is not there, something is wrong on our side.